src/pinghelper.cpp

```diff
@@ -47,8 +48,13 @@ void PingHelper::start(const QString& serverIpv4Gateway,
   // we happen to be on one of these unlucky devices, create a DnsPingSender
   // instead.
   if (!m_pingSender->isValid()) {
+    logger.warning() << "PingSenderFactory is not valid, trying DnsPingSender.";
     m_pingSender->deleteLater();
     m_pingSender = new DnsPingSender(m_source, this);
+    if (!m_pingSender->isValid()) {
+      logger.error() << "DnsPingSender is also not valid, using DummyPingSender.";
+      m_pingSender = new DummyPingSender(m_source, this);
```

Comment:
NIT: delete the now invalid dnspingsender :)

Reply:
Good catch! Done.